nearpy/storage/storage_memory.py

```diff
@@ -50,6 +50,17 @@ def store_vector(self, hash_name, bucket_key, v, data):
             self.buckets[hash_name][bucket_key] = []
         self.buckets[hash_name][bucket_key].append((v, data))
 
+    def store_many_vectors(self, hash_name, bucket_keys, vs, data):
+        """
+        Store a batch of vectors.
+        Stores vector and JSON-serializable data in bucket with specified key.
+        """
+        for idx, v in enumerate(vs):
```

Comment:
I suggest
```
if data is None:
    data = itertools.repeat(data)
for v, k, d in zip(vs, bucket_keys, data):
    self.store_vector(hash_name, k, v, d)
```

Reply:
By the way you can use that code as default implementation at the base `Storage` class. I am not sure if this is a good idea though.

Reply:
I have revised the storage_memory following your suggestion.
However, I think it may not be good to implement this as default implementation at the base Storage. At least we find that  Redis can use a more natural way to achieve it more efficiently. Leaving store_many_vectors() as an abstract method may force future contributors to think about efficient ways when implementing for other storage options.

Reply:
@xieqihui I agree. Let's keep it an abstract method.